engine/access/rest/http/routes/scripts.go

```diff
@@ -3,34 +3,85 @@ package routes
 import (
 	"github.com/onflow/flow-go/access"
 	"github.com/onflow/flow-go/engine/access/rest/common"
-	"github.com/onflow/flow-go/engine/access/rest/common/models"
+	commonmodels "github.com/onflow/flow-go/engine/access/rest/common/models"
+	"github.com/onflow/flow-go/engine/access/rest/http/models"
 	"github.com/onflow/flow-go/engine/access/rest/http/request"
+	accessmodel "github.com/onflow/flow-go/model/access"
 	"github.com/onflow/flow-go/model/flow"
 )
 
 // ExecuteScript handler sends the script from the request to be executed.
-func ExecuteScript(r *common.Request, backend access.API, _ models.LinkGenerator) (interface{}, error) {
-	req, err := request.GetScriptRequest(r)
+func ExecuteScript(r *common.Request, backend access.API, _ commonmodels.LinkGenerator) (interface{}, error) {
+	req, err := request.NewGetScript(r)
 	if err != nil {
 		return nil, common.NewBadRequestError(err)
 	}
 
+	executionState := req.ExecutionState
+	includeExecutorMetadata := executionState.IncludeExecutorMetadata
+
+	// TODO(mainnet #): remove this conditional: [TODO(Uliana): create an issue]
+	// "legacyParams" is only to temporarily support current behaviour.
+	// In the next spork, we should update this to always return an ExecuteScriptResponse.
+	legacyParams := executionState.AgreeingExecutorsCount == 0 &&
+		len(executionState.RequiredExecutorIDs) == 0 &&
+		!includeExecutorMetadata
```

Comment:
I think you can simplify this to just `legacyParams := !includeExecutorMetadata`.

we only need to make the backwards incompatible change in the case that the user wants the executor metadata.